script.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * SmartModalDialog creates a dialog with a group of buttons as select
+ * options. It also includes open/close animation and bubbling animation.
+ * The template is shown below:
+ *   <smart-dialog class="modal-dialog">
+ *     <div class="outer-container">
+ *       <div class="container">
+ *         <div class="modal-dialog-message">
+ *         </div>
+ *         <div class="modal-dialog-button-group">
+ *           <smart-button></smart-button>
+ *           ...
+ *           <smart-button></smart-button>
+ *         </div>
+ *       </div>
+ *     </div>
+ *   </smart-dialog>
+ */
+
+(function(exports) {
+  'use strict';
+
+  function SmartModalDialog() {
+
+    // Determine whether this dialog is opened or not
+    this.isOpened = false;
+    this._clickedIndex = null;
+
+    this.smartBubble = document.createElement('smart-bubbles');
+    document.body.appendChild(this.smartBubble);
+
+    this.element = document.createElement('smart-dialog');
+    this.element.classList.add('modal-dialog');
+    // make dialog focusable in order to catch focus from mouse click or touch
+    // event.
+    this.element.setAttribute('tabIndex', '-1');
+
+    // in order to make vertical align, we need additional container
+    this.outerContainer = document.createElement('div');
+    this.outerContainer.classList.add('outer-container');
+
+    this.container = document.createElement('div');
+    this.container.classList.add('container');
+
+    this.messageElement = document.createElement('div');
+    this.messageElement.classList.add('modal-dialog-message');
+
+    this.buttonGroup = document.createElement('div');
+    this.buttonGroup.classList.add('modal-dialog-button-group');
+    this.buttonGroup.setAttribute('smart-bubbles', 'true');
+
+    this.container.appendChild(this.messageElement);
+    this.container.appendChild(this.buttonGroup);
+
+    this.outerContainer.appendChild(this.container);
+    this.element.appendChild(this.outerContainer);
+    document.body.appendChild(this.element);
+
+    this.element.addEventListener('opened', this);
+    this.element.addEventListener('closed', this);
+    this.smartBubble.addEventListener('all-items-bubbled', this);
+  }
+
+  var proto = {};
+
+  proto.open = function(message, buttonSettings, onCancel) {
```

Comment:
We should expose an event listener with `onCancel`. But it's fine for this version. We should add it in the future.